doc/Makefile.am

```diff
@@ -99,14 +99,44 @@ doxygen_sources = \
 	$(top_srcdir)/tagmanager/src/tm_source_file.[ch] \
 	$(top_srcdir)/tagmanager/src/tm_workspace.[ch]
 
-Doxyfile.stamp: Doxyfile $(doxygen_sources)
-	$(AM_V_GEN)$(DOXYGEN) Doxyfile && echo "" > $@
+# set WARN_IF_UNDOCUMENTED because apparently doxygens warns for undocumented stuff
+# in headers (even though it's correctly documented in the corresponding .c file) only
+# for xml output
+Doxyfile-gi: Doxyfile
+	$(AM_V_GEN)$(SED) \
+		-e 's,addtogir=@internal,addtogir=,' \
+		-e 's,^\(GENERATE_HTML.*\)YES,\1NO,' \
+		-e 's,^\(GENERATE_XML.*\)NO,\1YES,' \
+		-e 's,^\(WARN_IF_UNDOCUMENTED.*\)YES,\1NO,' \
+		-e 's,^\(SORT_MEMBER_DOCS.*\)YES,\1NO,' \
+		-e 's,^\(SORT_BRIEF_DOCS.*\)YES,\1NO,' \
+		$< > $@ || { $(RM) $@ && exit 1; }
 
-all-local: Doxyfile.stamp
+Doxyfile.stamp: Doxyfile Doxyfile-gi $(doxygen_sources)
+	$(AM_V_GEN)$(DOXYGEN) Doxyfile-gi && $(DOXYGEN) Doxyfile && echo "" > $@
+
+ALL_TARGETS = Doxyfile.stamp
+
+if WITH_PYTHON
+
+geany-scintilla-gtkdoc.h: geany-gtkdoc.h
+
+geany-gtkdoc.h: Doxyfile.stamp $(top_srcdir)/scripts/gen-api-gtkdoc.py
+	$(AM_V_GEN)$(top_srcdir)/scripts/gen-api-gtkdoc.py xml -d $(builddir) \
+	-o geany-gtkdoc.h --sci-output geany-scintilla-gtkdoc.h
```

Comment:
Seems unecessary to me? Is this just to make sure *.h dont exist if the script failed?

Apart from that, the script can only fail early. It can't fail between the two output files (execpt file system errors).